src/Nethermind/Nethermind.HealthChecks/NodeHealthService.cs

```diff
@@ -69,30 +77,54 @@ public CheckHealthResult CheckHealth()
             long netPeerCount = _syncServer.GetPeerCount();
             SyncingResult syncingResult = _ethSyncingInfo.GetFullInfo();
 
-            if (_isMining == false && syncingResult.IsSyncing)
+            if (_api.SpecProvider!.TerminalTotalDifficulty != null)
             {
-                AddStillSyncingMessage(messages, syncingResult);
+                if (syncingResult.IsSyncing)
+                {
+                    AddStillSyncingMessage(messages, syncingResult);
+                }
+                else
+                {
+                    AddFullySyncMessage(messages);
+                }
                 CheckPeers(messages, netPeerCount);
+
+                bool clAlive = CheckClAlive();
+
+                if (!clAlive)
+                {
+                    AddClUnavailableMessage(messages);
+                }
+
+                healthy = !syncingResult.IsSyncing & clAlive;
             }
-            else if (_isMining == false && syncingResult.IsSyncing == false)
-            {
-                AddFullySyncMessage(messages);
-                bool peers = CheckPeers(messages, netPeerCount);
-                bool processing = IsProcessingBlocks(messages);
-                healthy = peers && processing;
-            }
-            else if (_isMining && syncingResult.IsSyncing)
-            {
-                AddStillSyncingMessage(messages, syncingResult);
-                healthy = CheckPeers(messages, netPeerCount);
-            }
-            else if (_isMining && syncingResult.IsSyncing == false)
+            else
             {
-                AddFullySyncMessage(messages);
-                bool peers = CheckPeers(messages, netPeerCount);
-                bool processing = IsProcessingBlocks(messages);
-                bool producing = IsProducingBlocks(messages);
-                healthy = peers && processing && producing;
+                if (!_isMining && syncingResult.IsSyncing)
+                {
+                    AddStillSyncingMessage(messages, syncingResult);
+                    CheckPeers(messages, netPeerCount);
+                }
+                else if (!_isMining && !syncingResult.IsSyncing)
```

Comment:
why we're not checking if CL is alive if the node is not syncing?

Reply:
It's legacy code, if merge is not enabled